### PR TITLE
Incorporate changes to AWS stack based on feedback

### PR DESCRIPTION
**Pull request by phildier**
- Include IAM permissions used to create the stack based on CloudTrail logs
- Include AWS-specific codecov.yml.example
- Update default instance types
- Adds notes on instance type requirements
- Adds notes on Traefik
- Allow Traefik to be disabled
- Adds notes on granting firewall access to the stack
- Bump Codecov version to 4.5.0
- Add statsd port env var

### CODE REVIEW

File: aws_elastic_kubernetes_service/codecov.yml.example [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  enterprise_license: "Codecov-Enterprise-License-Key"
+  http:
+    cookie_secret: "cookie-secret"
+github:
+  client_id: "example-client-id"
+  client_secret: "example-client-secret"
+services:
+  ci_providers:
+  minio:
+    client_uploads: true
```
Comment (hootener) [inline aws_elastic_kubernetes_service/codecov.yml.example:12]: are these the minio settings that get applied, or will the S3 bucket configuration get applied instead?

Comment (phildier) [reply aws_elastic_kubernetes_service/codecov.yml.example:12]: I'm not clear on what you're asking here, but the s3 bucket and credentials are passed to web and worker through env vars.  I could pass this setting as well and avoid the minio section in the yml completely if you think that would be clearer.

Comment (hootener) [reply aws_elastic_kubernetes_service/codecov.yml.example:12]: The use of env vars is :+1: . However, `client_uploads:true` likely shouldn't be used in the k8s configuration if cloud services are being used. 

If you upload with this stack is the upload url that's returned `/upload/v4` or `/upload/v2` ?

Comment (phildier) [reply aws_elastic_kubernetes_service/codecov.yml.example:12]: Oh, OK.  I thought that was necessary for minio when using the sdk.  I will remove that and test again to be sure it still works.

Comment (hootener) [reply aws_elastic_kubernetes_service/codecov.yml.example:12]: @phildier did this work without the `client_uploads: true` specification?

**Pull request merged**